pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -243,5 +249,9 @@ func (p *Provider) getVnetInfoLabels(subnetID string) (map[string]string, error)
 		azureCNIOverlayLabel: strconv.FormatBool(true),
 		podNetworkTypeLabel:  consts.NetworkPluginModeOverlay,
 	}
+
+	parsedVersion, _ := semver.Parse(kubernetesVersion)
+	vnetLabels[networkStatelessCNILabel] = lo.Ternary(parsedVersion.Minor >= 34, "true", "false")
```

Comment:
Again, maybe best following the style on version comparison from `UseAzureLinux3`?
https://github.com/Azure/karpenter-provider-azure/blob/38368736f1304b6156443367bcc9ced16cdcd006/pkg/providers/imagefamily/utils.go#L32

In this case it would be:
```suggestion
	vnetLabels[networkStatelessCNILabel] = lo.Ternary(parsedVersion.GE(semver.Version{Major: 1, Minor: 34}), "true", "false")
```